src/main/java/fi/aalto/cs/apluscourses/intellij/actions/CourseProjectAction.java

```diff
@@ -280,6 +289,25 @@ private URL tryGetCourseUrl(@NotNull Project project) {
     }
   }
 
+  private CourseItemViewModel[] fetchCourses(@NotNull Project project) {
+    try {
+      var url = new URL(COURSE_LIST_URL);
+      var courseStream = CoursesClient.fetch(url);
+      var yaml = new Yaml();
+
+      @SuppressWarnings("unchecked") var courseList = (List<Map<String, String>>) yaml.load(courseStream);
```

Comment:
This looks "non-object-oriented" but from what I've found it's the simplest way. Other methods that look "cleaner" use some ClassLoaders or a metric ton of boilerplate code.